vocabs/DID-core-shape.ttl

```diff
@@ -0,0 +1,129 @@
+# You can run this, for example, with:
+#  pyshacl -s DID-core-shape.ttl -e DID-core.ttl -f human test.ttl
+#
+
+@prefix did:  <https://w3id.org/did#> .
+@prefix as:   <https://www.w3.org/ns/activitystreams> .
+@prefix rdf:  <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix xsd:  <http://www.w3.org/2001/XMLSchema#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix sh:   <http://www.w3.org/ns/shacl#> .
+@base <https://www.w3.org/ns/did/core-vocabulary> .
+
+
+did:DIDDocumentShape a sh:NodeShape ;
+    # Some properties are valid only on DID Documents and they will set the right shape
+    # Note that this is not the case for did:verificationMethod (which can also be used
+    # on a did:ServiceEndpoint) or did:controller (that can also be used on did:VerificationMethod)
+
+    sh:targetClass did:DIDDocument ;
+    sh:targetSubjectsOf
+        did:alsoKnownAs,
```

Comment:
consider adding canonicalId / equivalentId here cc @csuwildcat

Reply:
Those are not in the core vocabulary of the DID specification, they are resolution metadata properties. *If* we introduced them in the core vocabulary, that means they could be used in a DID Document, and I believe the WG firmly voted agaist that (sorry @csuwildcat...)

Reply:
I'm not sure what this means, but I hope it doesn't mean the properties are being removed from the spec, still have the same gravity, and include the same normative MUST statements.

Reply:
@csuwildcat 

> I'm not sure what this means, but I hope it doesn't mean the properties are being removed from the spec, still have the same gravity, and include the same normative MUST statements.

To put your mind at ease: no, it doesn't.

The vocabulary, as well as the SHACL constraint language, is only relevant to the [core terms in the model](https://w3c.github.io/did-core/#core-properties). To be even more precise: the vocabulary is of relevance if and only if the DID Document is expressed in JSON-LD and the RDF mapping of that JSON-LD is of use by a specific application (method or otherwise). All this completely orthogonal to the resolution metadata properties, their specification, and usage.